openqaoa/backends/simulators/qaoa_analytical_sim.py

```diff
@@ -0,0 +1,111 @@
+
+# not sure about those ones:
+from ...qaoa_parameters.baseparams import QAOACircuitParams, QAOAVariationalBaseParams 
+from typing import Union, List, Tuple, Type, Optional
+
+
+class QAOABackendAnalyticalSimulator(QAOABaseBackend):
+    r"""
+    A simulator class, specific for QAOA with a single layer, p=1, starting with a layer of Hadamards and using the X mixer. 
+    Works by calculating the expectation value of the given quantum circuit (specificied with beta and gamma angles) from the analytical formula derived in arXiv:2011.13420v2.
+     
+     Parameters
+    ----------
+    circuit_params: QAOACircuitParams
+        An object of the class ``QAOACircuitParams`` which contains information on 
+        circuit construction and depth of the circuit. Note that it only works for p=1 and the X Mixer Hamiltonian.
+    """
+    
+    def __init__(self, circuit_params: QAOACircuitParams, 
+                 prepend_state = None,
+                 append_state = None,
+                 init_hadamard = True,
+                 cvar_alpha = 1):
+        
+        QAOABaseBackend.__init__(self, circuit_params,
+                                         prepend_state = None,
+                                         append_state = None,
+                                         init_hadamard = True,
+                                         cvar_alpha = 1)
+        
+        self.measurement_outcomes = {}  # passing empty dict for the logger since measurements are irrelevant for this backend.
+        
+        
+        # check if conditions for the analytical formula are met
+        assert self.circuit_params.p == 1, "Analytical formula only holds for p=1."
+        
+        for gatemap in self.circuit_params.mixer_qubits_singles:
+            assert gatemap == 'RXGateMap', "Analytical formula only holds for X mixer."
+        
+    def assign_angles(self):
+        raise NotImplementedError("This method is irrelevant for this backend")
+        
+    def obtain_angles_for_pauli_list(self):
+        raise NotImplementedError("This method is irrelevant for this backend")
+        
+    def qaoa_circuit(self):
+        raise NotImplementedError("This method is irrelevant for this backend")
+        
+    def get_counts(self):
+        raise NotImplementedError("This method is irrelevant for this backend")
+        
+    def expectation(self, params: QAOAVariationalBaseParams) -> float:
+        """
+        Compute the expectation value w.r.t the Cost Hamiltonian analytically.
+
+        Parameters
+        ----------
+        params: `QAOAVariationalBaseParams`
+            The QAOA parameters - an object of one of the parameter classes, containing 
+            variable parameters.
+
+        Returns
+        -------
+        float:
+            Expectation value of cost operator wrt to the QAOA parameters according to the analytical expression for p=1.
+        """
+        betas = params.betas
```

Comment:
This is going to work only for the standard parametrisation. Is this wanted?

Reply:
So, each parameterization has `betas` and `gammas` attributes. Since the expectation value formula only depends on `betas`, `gammas`, one can extract these values from any parameterization to plug into the formula and obtain the cost function.

Reply:
@kidiki, one thing to ensure is what happens in the case of `Extended` (including `FourierExtended`) and `WithBias` parameterizations, which may not have exactly these attributes but rather `gammas_singles`, `gammas_pairs`, `betas_singles`, `betas_pairs`

Reply:
Is it enough to assert that the params must be standard for now? and then later we can re-derived the analytical formula for extended?

Reply:
AFAIK, the generalization to Extended shouldn't be too hard. Let's make an attempt (perhaps @ezerc can help here), and if that's not the case then we restrict it to support only Standard.